File: python/ais_sdk/image_antiporn.py
# -*- coding:utf-8 -*-

import sys
import json
import ais_sdk.ais as ais
import ais_sdk.utils as utils
import ais_sdk.signer as signer

def request_moderation_url(token, inner_path, image_str=None, url=None):
    endpoint = utils.get_endpoint(ais.AisService.MODERATION_SERVICE)
    _url = 'https://' + endpoint + inner_path

    if sys.version_info.major >= 3:
        if image_str != '':
            image_str = image_str.decode("utf-8")

    _data = {
        "image": image_str,
        "url": url
    }

    status_code, resp = utils.request_token(_url, _data, token)
    if sys.version_info.major < 3:
        return resp
    else:
        return resp.decode('utf-8')


#
# access moderation, image anti-porn,post data by token
#
def image_antiporn(token, image_str=None, url=None):
    print(request_moderation_url(token, '/v1.1/moderation/image/anti-porn', image_str, url))
    print(request_moderation_url(token, '/v1.0/moderation/image', image_str, url))


def request_moderation_url_aksk(sig, inner_path, image_str=None, url=None):
    endpoint = utils.get_endpoint(ais.AisService.MODERATION_SERVICE)
    _url = 'https://' + endpoint + inner_path

    if sys.version_info.major >= 3:
        if image_str != '':
            image_str = image_str.decode("utf-8")

    _data = {
        "image": image_str,
        "url": url
    }

    kreq = signer.HttpRequest()
    kreq.scheme = "https"
    kreq.host = endpoint
    kreq.uri = inner_path
    kreq.method = "POST"
    kreq.headers = {"Content-Type": "application/json"}
    kreq.body = json.dumps(_data)

    status_code, resp = utils.request_aksk(sig, kreq, _url)
    if sys.version_info.major < 3:
        return resp
    else:
        return resp.decode('utf-8')

#
# access moderation, image anti-porn,post data by ak,sk
#
def image_antiporn_aksk(_ak, _sk, image_str=None, url=None):
    sig = signer.Signer()
    sig.AppKey = _ak
    sig.AppSecret = _sk
    print (request_moderation_url_aksk(sig, '/v1.1/moderation/image/anti-porn', image_str, url))
    print (request_moderation_url_aksk(sig, '/v1.0/moderation/image', image_str, url))

File: javascript/ais_sdk/image_antiporn.js
var https = require("https");
var utils = require("./utils");
var signer = require("./signer");
var ais = require("./ais");

module.exports = {
    image_antiporn: function (token, data, url, callback) {

        var endPoint = utils.getEndPoint(ais.MODERATION_SERVICE);

        // 构建请求信息和请求参数信息
        var requestData = {"image": data, "url": url};
        var requestBody = JSON.stringify(requestData);

        var host = endPoint;
        var method = "POST";
        var uri = ais.IMAGE_ANTI_PORN;
        var headers = {"Content-Type": "application/json", "X-Auth-Token": token, "Content-Length": requestBody.length};
        var options = utils.getHttpRequestEntityOptions(host, method, uri, headers);

        var request = https.request(options, function (response) {

            // 验证服务调用返回的状态是否成功，如果为200, 为成功, 否则失败。
            if (response.statusCode !== 200) {
                console.log('Http status code is: ' + response.statusCode);
            }
            // 处理结果信息，输入返回信息
            response.on("data", function (chunk) {
                callback(chunk.toString())
            })
        });

        request.on("error", function (err) {
            console.log(err.message);
        });

        request.write(requestBody);
        request.end();
    },

    image_antiporn_aksk: function (_ak, _sk, data, url, callback) {

        // 配置ak，sk信息
        var sig = new signer.Signer();
        sig.AppKey = _ak;                   // 构建ak
        sig.AppSecret = _sk;                // 构建sk

        var endPoint = utils.getEndPoint(ais.MODERATION_SERVICE);

        // 构建请求信息和请求参数信息
        var requestData = {"image": data, "url": url};
        var _headers = {"Content-Type": "application/json"};
        var req = new signer.HttpRequest();
        var options = utils.getHttpRequestEntity(sig, req, endPoint, "POST", ais.IMAGE_ANTI_PORN, "", _headers, requestData);

        var requset = https.request(options, function (response) {

            // 验证服务调用返回的状态是否成功，如果为200, 为成功, 否则失败。
            if (response.statusCode !== 200) {
                console.log('Http status code is: ' + response.statusCode);
            }
            response.on("data", function (chunk) {
                callback(chunk.toString());
            })
        });

        requset.on("error", function (err) {
            console.log(err.message);
        });

        requset.write(req.body);
        requset.end();
    }

};